src/Cake.Core/Configuration/Parser/ConfigurationParser.cs

```diff
@@ -37,7 +37,7 @@ public ConfigurationParser(IFileSystem fileSystem, ICakeEnvironment environment)
             using (var stream = file.OpenRead())
             using (var reader = new StreamReader(stream))
             {
-                return Read(reader.ReadToEnd());
+                return Read(EnvironmentVariableSubstitution.Substitute(_environment, reader.ReadToEnd()));
```

Comment:
I took a cursory look at your `EnvironmentVariableSubstitution.Substitute` and it seems that it's re-implementing an existing .NET feature -> [`Environment.ExpandEnvironmentVariables`](https://docs.microsoft.com/en-us/dotnet/api/system.environment.expandenvironmentvariables?view=net-5.0) is that correct?

If that's true, could we use the built-in function instead? Otherwise, what's different/better about this implementation?

Reply:
Oh.My.Lord.  Well, this is awkward... 
I've been programming .net for the last twenty years, and I wasn't aware that the framework had the functionalty built-in.
Oh well, at least I learned something by doing this 😁 

I'll fix this, I think using built-in functionality would be better.

Reply:
Rewritten to use built-in functionality instead